speechbrain/lobes/models/transformer/TransformerASR.py

```diff
@@ -187,9 +187,9 @@ def forward(self, src, tgt, wav_len=None, pad_idx=0):
             # use standard sinusoidal pos encoding in decoder
             tgt = tgt + self.positional_encoding_decoder(tgt)
             # FIXME we use pos embs also on enc output
-            encoder_out = encoder_out + self.positional_encoding_decoder(
-                encoder_out
-            )
+            # encoder_out = encoder_out + self.positional_encoding_decoder(
+            #    encoder_out
+            # )
             pos_embs_encoder = None  # self.positional_encoding(src)
             pos_embs_target = None
```

Comment:
`positional_encoding_decoder` is only set for `attention_type == "RelPosMHAXL"`
```python
        if attention_type == "RelPosMHAXL":
            self.positional_encoding = RelPosEncXL(d_model)
            self.positional_encoding_decoder = PositionalEncoding(
                d_model, max_length
            )
```

So - what was this fixme about in the first place?

---

btw - it's also used in G2P:
```
grep -r positional_encoding_decoder speechbrain
```

---

What is this line about?
```python
pos_embs_encoder = None # self.positional_encoding(src)
pos_embs_encoder = self.positional_encoding(src)  # if we drop the other one, as you suggested
```
Maybe the comment there can be dropped (it would relate to the forward in `class RelPosMHAXL(nn.Module)`).
```python
...
self.linear_pos = nn.Linear(embed_dim, embed_dim, bias=False)
...
p_k = self.linear_pos(pos_embs).view(
    1, -1, self.num_heads, self.head_dim
)
```